compiler/rustc_middle/src/ty/adt.rs

```diff
@@ -249,15 +249,25 @@ impl Into<DataTypeKind> for AdtKind {
 }
 
 impl AdtDefData {
+    pub(super) fn new_from_flags(
+        did: DefId,
+        variants: IndexVec<VariantIdx, VariantDef>,
+        flags: AdtFlags,
+        repr: ReprOptions,
+    ) -> Self {
+        AdtDefData { did, variants, flags, repr }
+    }
+
     /// Creates a new `AdtDefData`.
     pub(super) fn new(
         tcx: TyCtxt<'_>,
-        did: DefId,
+        did: LocalDefId,
```

Comment:
Setting this to local so we enforce that `AdtDefData::new` is called only on local def ids.